sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/LogicalQueryStage.scala

```diff
@@ -54,4 +54,6 @@ case class LogicalQueryStage(
     }
     physicalStats.getOrElse(logicalPlan.stats)
   }
+
+  override def maxRows: Option[Long] = stats.rowCount.map(_.min(Long.MaxValue).toLong)
```

Comment:
If there is other physical nodes above `QueryStageExec` the stats is not accurate. Seems the only node can exist here is aggregate, so the maxRows is still accurate.

Reply:
We should trust the existing framework. If the `maxRows` stats can be wrong, then `EliminateLimits` is also wrong even without AQE.

I don't think we need to highlight `LogicalQueryStage` in the doc of `EliminateLimits`. We just need to follow the existing framework and make sure the `maxRows` is not under-estimated.

Reply:
OK, cleaned the comment of `EliminateLimits `